### PR TITLE
[fix](coldhot) fix cannot cancel storage policy of partition

### PR DESCRIPTION
**Pull request by Yulei-Yang**
Issue:
execute a stmt to cancel a partition's storage policy returns ok, but it does not work:
image

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

**Review by lide-reed — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java [hunk 1]
```diff
@@ -764,6 +764,12 @@ public void modifyPartitionsProperty(Database db,
                 // check currentStoragePolicy resource exist.
                 Env.getCurrentEnv().getPolicyMgr().checkStoragePolicyExist(currentStoragePolicy);
                 partitionInfo.setStoragePolicy(partition.getId(), currentStoragePolicy);
+            } else {
+                if (partition.getRemoteDataSize() > 0) {
+                    throw new AnalysisException(
+                        "Cannot cancel storage policy for partition which is already on code storage.");
```
Comment (xy720) [inline fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:770]:
```suggestion
                        "Cannot cancel storage policy for partition which is already on cold storage.");
```

**Review by xy720 — COMMENTED** [review]
Are you sure that canceling the partition's policy will not cause any other issues?

**Comment by xy720** [comment]
We should not cancel any storage migration policies to prevent from being unable to recover data if the policies are canceled after data migration.

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java (hunk 1 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java]: done

**Comment by Yulei-Yang** [comment]
run buildall

**Review by xy720 — DISMISSED** [review]
LGTM

**Comment by Yulei-Yang** [comment]
run buildall

**Review by xy720 — APPROVED** [review]

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**